custom_components/oasis_mini/pyoasiscontrol/clients/http_client.py

```
@@ -0,0 +1,349 @@
+    async def async_get_mac_address(self, device: OasisDevice) -> str | None:
+        """
+        Fetch the device MAC address using the device's HTTP GETMAC endpoint.
+
+        Returns:
+            str: The MAC address with surrounding whitespace removed, or `None` if it could not be retrieved.
+        """
+        try:
+            mac = await self._async_get(params={"GETMAC": ""})
+            if isinstance(mac, str):
+                return mac.strip()
+        except Exception:  # noqa: BLE001
```

Ruff (0.14.5)

Unused `noqa` directive (unused: `BLE001`)

Remove unused `noqa` directive

(RUF100)